open_seq2seq/data/speech2text/speech_utils.py

```diff
@@ -399,6 +400,22 @@ def get_speech_features_librosa(signal, sample_freq, num_features,
   std_dev = np.std(features, axis=norm_axis)
   features = (features - mean) / std_dev
 
+  if augmentation:
+    n_freq_mask = augmentation.get('n_freq_mask', 0)
+    n_time_mask = augmentation.get('n_time_mask', 0)
+    width_freq_mask = augmentation.get('width_freq_mask', 10)
+    width_time_mask = augmentation.get('width_time_mask', 50)
+
+    for idx in range(n_freq_mask):
+      freq_band = np.random.randint(width_freq_mask + 1)
+      freq_base = np.random.randint(0, features.shape[1] - freq_band)
+      features[:, freq_base:freq_base+freq_band] = 0
+    for idx in range(n_time_mask):
+      time_band = np.random.randint(width_time_mask + 1)
+      if features.shape[0] - time_band > 0:
```

Comment:
Do we need a similar check for freq_band?
Shouldn't we also add an else that drops the entire spectrogram?

Reply:
Frequency shape is fixed (that is, `num_audio_features`). Time duration varies from utterance to utterance. I'd vote for skipping masking in that case.